.github/workflows/codeql.yml

```diff
@@ -0,0 +1,32 @@
+name: "CodeQL"
+
+on:
+  push:
+    branches: [master]
+  pull_request:
+    branches: [master]
+    paths-ignore:
+      - '**/*.md'
+      - '**/*.txt'
+      - '**/*.rst'
```

Comment:
Is this really needed?

Reply:
Not needed, but are there reasons to do codeQL checks if we just update the readme or other text files?

Reply:
(I would on the other hand propose to add this to general build files to avoid all CI checks for a doc update.)

Reply:
Ok, makes sense.